cdap-common/src/main/java/io/cdap/cdap/common/operation/LongRunningOperation.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.operation;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.cdap.cdap.proto.operationrun.OperationError;
+import io.cdap.cdap.proto.operationrun.OperationMeta;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+/**
+ * LongRunningOperation represents a long-running asynchronous operation.
+ *
+ * @param <T> Type of operation request
+ */
+public interface LongRunningOperation<T> {
+
+  /**
+   * Get the {@link Type} T of the request.
+   */
+  Type getRequestType();
+
+  /**
+   * Run the operation with the given request.
+   *
+   * @param request the operation input
+   * @param updateMetadata func to update the metadata of the operation. This would be passed by
+   *     the runner.
+   * @return {@link ListenableFuture} containing the {@link OperationError} for the run
+   */
+  ListenableFuture<OperationError> run(T request, Consumer<OperationMeta> updateMetadata)
+      throws Exception;
```

Comment:
Why declare throws Exception? When will it throw? Generally for method that returns a future, it shouldn't throw, but rather return a immediate failure future if there is error before running the async task. This makes the caller error handling in one place rather than scattered into two.